Libplanet/Blocks/Block.cs

```diff
@@ -118,6 +118,20 @@ public Block(Bencodex.Types.Dictionary dict)
         {
         }
 
+        public Block(Block<T> block, HashDigest<SHA256>? actionHash)
+            : this(
+                block.Index,
+                block.Difficulty,
+                block.TotalDifficulty,
+                block.Nonce,
+                block.Miner,
+                block.PreviousHash,
+                block.Timestamp,
+                block.Transactions)
+        {
+            ActionHash = actionHash;
+        }
+
         private Block(RawBlock rb)
```

Comment:
Though I can't grab all the lines of the private constructor `Block(RawBlock)`, I think there will be a problem which stales actionHash. Now `Block.ActionHash` can be set through only the constructor `Block(Block<T>, HashDigest<SHA256>?)`, but it doesn't use that. So it may stale the `ActionHash` value after serialize-and-deserialize, and it will be an invalid block. Also, I believe it will be better if there is a test for the cases.

PS. If I missed seeing another line which set the value, please let me know it 😀

Reply:
Yes, you're right. Actually I notice that after PR open. 😓 We need a test about convert between `RawBlock` <-> `Block`

Reply:
Fit it at 1c1371ae7f6b520cacccbd3752ff88d2be29c77e